pom.xml

```diff
@@ -35,6 +35,17 @@
 
 
     <dependencies>
+        <!-- logging please dont move loogin further down - order of dependencies does matter-->
+        <dependency>
+            <groupId>com.github.ONSdigital</groupId>
+            <artifactId>dp-logging</artifactId>
+            <version>${dp-logging.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
```

Comment:
How come this new dependency is needed? Is it so that you can use `com.google.common.io.ByteStreams`? If so, is there an alternative that does not require pulling a whole new library?

Reply:
This is the solution I found that goes with java 1.8 . Java 1.11 would have its own NullOutPutStream.

Reply:
Was this done because `NullOutputStream` is deprecated? If so, it looks like you could simply use `NullOutputStream.INSTANCE` instead. E.g.:

```java
ShaOutputStream output = new ShaOutputStream(NullOutputStream.INSTANCE)) {
```

Does that work?

Reply:
yes, this was the only reason. Maybe you are right. But just reading the code INSTANCE uses the deprecated constructor. Not sure this will improve the situation?

public class NullOutputStream extends OutputStream {
    public static final NullOutputStream INSTANCE = new NullOutputStream();
    /** @deprecated */
    @Deprecated
    public static final NullOutputStream NULL_OUTPUT_STREAM;

    /** @deprecated */
    @Deprecated
    public NullOutputStream() {
    }

Reply:
You're right in that `INSTANCE` uses that constructor, but that's not a problem, in fact the constructor's Javadoc says that you should use `INSTANCE` because the constructor will be private in the future:

```java
    /**
     * Deprecated in favor of {@link #INSTANCE}.
     *
     * TODO: Will be private in 3.0.
     *
     * @deprecated Use {@link #INSTANCE}.
     */
    @Deprecated
    public NullOutputStream() {
    }
```

Reply:
👍 yes, makes sense.